src/core/create_compilers/RollupCompiler.ts

```diff
@@ -167,37 +169,45 @@ export default class RollupCompiler {
 					// We need to avoid the entry chunk both here and below so that we don't walk everything
 					// We should remove the ciricular dependency in Sapper so that this isn't a concern
 					const transitive_deps = js_deps(tree_entry_chunk, {
-						filter: ctx => ctx.chunk.fileName !== tree_entry_chunk.fileName,
 						walk: ctx => !ctx.dynamicImport && ctx.chunk.fileName !== entry_chunk.fileName });
 					for (const chunk of transitive_deps) {
-						if (!processed_chunks.has(chunk)) {
-							handle_chunks(chunk, [chunk]);
+						if (!global_chunks.has(chunk)) {
+							handle_chunks(chunk, [chunk], subtree);
 						}
 					}
+				}
 
-					// Put everything that's leftover into the entry chunk in order to include css
-					// that is in a dynamically imported chunk
-					const chunks_for_tree = js_deps(tree_entry_chunk,
-						{ walk: ctx => !subtree || ctx.chunk.fileName !== entry_chunk.fileName });
-					const unused = new Set(chunks_for_tree.filter(x => !processed_chunks.has(x)));
-					handle_chunks(tree_entry_chunk, unused);
+				function isRoute(file_path: string) {
```

Comment:
`is_route` please, to keep with the snake case used elsewhere.

Reply:
done